### ISSUE TITLE
Mean Radiant Temperature Estimations

### ISSUE THREAD

**Issue by benterich** [CenterForTheBuiltEnvironment/pythermalcomfort#144]
**Is your feature request related to a problem? Please describe.**
Not really a problem, just something I would find useful. There are various ways to estimate Mean Radiant Temperature, hence it would be useful to have it in this package.

**Describe the solution you'd like**
One method that can estimate Mean Radiant Temperature. For example the "traditional method" by Fanger, "Six-directional method",  "Black globe thermometer method".

Preferably I would propose, one MeanRadiantTemperature Class, that holds the various methods and its inputs.

**Describe alternatives you've considered**
Otherwise, its also possible to create seperate methods for each estimation independantly.

I can inccoperate them :)!

**Comment by FedericoTartarini** [comment]
I completely agree that it would be nice to have equations to estimate the mean radiant temperature. 

There is already one function that allows you to do that, it is called [`solar_gain_on_people`](https://pythermalcomfort.readthedocs.io/en/latest/reference/pythermalcomfort.html#solar-gain-on-people). The name for this function is not great and we could consider a better naming convention. Moreover, the function to calculate the MRT from the globe temperature is also there already.

I agree that we should have one class/funcition and then people can choose the method they use. Please note that, I am in the process of updating all the function in pythermalcomfort, you can see how I am structuring them by looking at the `development` branch. I have not updated all the function yet, but you can have a look at the adaptive ASHRAE function. 

It would also be very useful to incorporate a function that estimates mean radiant temperature based on the solar radiant, cloud coverage. The authors of the Clima App, have already done that and they have described the method in [this paper](https://www.mdpi.com/1660-4601/18/21/11317).

Could you please help me find all the functions we can use to calculate the mean radiant temperature and then we can have a quick call in which we discuss how we are going to add them to pythermalcomfort?

**Comment by benterich** [comment]
Yes, i would love to help out!

I saw how you updated the functions and it seems clear to me. I am gonna have a look at the estimation done in the ClimaApp and gonna collect all to me known methods to estimate MRT. Then I can make a quick concept which we can dicsuss, ideal would be after the 2nd December, since then i have conducted my masterthesis defense.

Kind regards,
Ben

**Comment by FedericoTartarini** [comment]
@benterich Another article that is relevant is [this](https://publications.ibpsa.org/proceedings/bs/2019/papers/BS2019_210301.pdf) which compares the different equations used by different tools to calculate the mean radiant temperature. 

However, it should be noted that with pythermalcomfort we will not be able to calculate all the view factors and surface temperatures of the surrounding objects. We will need to make several assumption, for example, that the temperature of the surrounding surfaces is equal to the air temperature and the sky is absorbing heat from the human body. I have not read in detail the paper from the ClimApp but that should be a good starting point.

**Comment by benterich** [comment]
"we will not be able to calculate all the view factors and surface temperatures of the surrounding objects. We will need to make several assumption" - Yes i agree, we definitly have to, otherwise we basically building an engine. However, if we provide some smart and carefully though inputs, i think  there might be potential to  to adpat pythermalcomfort methods to more complicated calculations if necessary or integrate it into other workflows.

So far after some quick searches, beside various Softwares stating how they estimate, I havent found many papers discussion the various techniques to estimate mean radiant temperature, there are many that only calibrate measurement techniques, comparing it sometimes to various computed models.

I think, I try to work out some sort of mindmap with the various functions and correlations and their inputs. From there it should be good to built upon our functions.

**Comment by ruijis** [comment]
We can provide the linear or fourth power formula of MRT based on surface temperatures and their view factors. It is well documented in Wikipedia: [https://en.wikipedia.org/wiki/Mean_radiant_temperature](url)

I suggest including the fourth power formula in the pythermalcomfort. The linear formula is too simple, and people can directly code it and calculate the MRT within seconds.

People will need to calculate or assume view factors. I used to calculate them in a simple way based on the percentage of different surface areas.

**Comment by benterich** [comment]
Hello, apologies for my late reply. I haven’t made much progress yet as Christmas and the New Year turned out to be busier than expected. I like @ruiji's proposal. For the view factor estimation, does it make sense / is it useful to include emissivity, direction, and distance weighting? I think its fairly straight forward to program, if not available, we can set some standard paramters.

I also suggest adding a separate method for the Black Globe Thermometer approach since it’s quite commonly used, right?

I will prepare a first draft this week and let you guys know.

**Comment by ruijis** [comment]
The globe temperature method for mrt estimation is available in the pythermalcomfort

https://pythermalcomfort.readthedocs.io/en/latest/reference/pythermalcomfort.html#pythermalcomfort.psychrometrics.t_mrt

**Comment by FedericoTartarini** [comment]
Yes, we can definitely include the forth power formula of MRT. This is going to be a nice addition. To be honest, however, I don't think anyone will ever use it since it is extremely hard to correctly estimate the view factor.

I would have loved it also to add equations to estimated the mean temperature outdoors as we previously discussed.

**Comment by FedericoTartarini** [comment]
Hi @benterich I was talking with Bruce and he also suggested the following.


> Calculation of MRT from a weather model in the current method you are using will give a rough estimate (he is referring to the solar gain model I am using), but what's better would be combining the various radiative fluxes, both direct and diffuse (allowing for the influence of clouds, surface albedo etc etc) that are calculated as part of the model's physics.
Lucky that John pointed out the ECMWF pages at [https://charts.ecmwf.int/products/medium-thermofeel](https://url.au.m.mimecastprotect.com/s/hHDuCE8wmrt6joKnRtwhmS7aZGw?domain=charts.ecmwf.int) which has done just that!
Following the links on that page led me to [this link](https://url.au.m.mimecastprotect.com/s/tilDCGv0oyCL9ZoqjCpiqSBdY-6?domain=confluence.ecmwf.int) where you will find that the ECMWF model has recently added MRT as an output parameter in its operational model as of November 2024. 

> It's possible that MRT could be derived in a similar way for other weather models (like BoM's) using ECMWF's method (see attached paper) which could be useful for the pythermalcomfort library. Or maybe you can adapt what's in the  [thermofeel](https://url.au.m.mimecastprotect.com/s/bGlOCJyBrGfQoLPy0uLsPSyT1dd?domain=doi.org) Python library (available on [github](https://url.au.m.mimecastprotect.com/s/mxi9CK1DvKTrkx093cptDS5g2GZ?domain=github.com)).

He also recommended reading [this paper](https://doi.org/10.1007/s00484-020-01900-5) if you have not done it already

**Comment by jli113** [comment]
Page not found

**Comment by FedericoTartarini** [comment]
Hi @jli113 I am not sure which issue you are trying to report.
